furusiya/model/components/base.py

```diff
@@ -0,0 +1,17 @@
+class Component:
+    """
+    Basic class describing a component.
+
+    Since all components inherit from this, it shouldn't be passed to GameObject's get_component method;
+    lest it returns a random component.
+
+    Class Attributes:
+        component_type (str): Defines the component slot; used for defining mutually exclusive components
+
+    Attributes:
+        owner (GameObject): The object who this component belongs to
+    """
+    component_type = None
```

Comment:
This should be just the component class. I understand that it introduces complexity (what if I add both an `AnimalComponent` and a `CatComponent(AnimalComponent)`?  Well, checking the exact type gets around that), but it's much more intuitive and doesn't require additional strings/enums/values when we add new components. 

Please please pretty-please fix this.

Reply:
Consider the following:
- add an AI extension to a GameObject, say, BasicMonster.
- add another AI extension of a different type. e.g: ConfusedMonster.

The latter won't override the former, and instead, one of them will be picked up; whichever is lucky enough to be in the first index in `possible_matches`.

Possible solutions?
- The fix I already made
- One class per component type. But I trust you can see how this will be very limiting, unless we dynamically create components from functions or something.